torch/overrides.py

```diff
@@ -1766,16 +1766,16 @@ def is_tensor_like(inp):
     return type(inp) is torch.Tensor or hasattr(type(inp), "__torch_function__")
 
 
+_cur_torch_function_mode = []
+
 def _wrap_torch_function(f):
     @functools.wraps(f)
     def wrapped(self, *args, **kwargs):
         if isinstance(f, classmethod):
             raise RuntimeError("TorchFunctionMode's torch_function function " +
                                "should be a normal method not a class method")
-        inner = getattr(self, "inner", None)
 
-        with enable_torch_function_mode(inner):
-            return f(self, *args, **kwargs)
+        return f(self, *args, **kwargs)
```

Comment:
Ok yeah I see the popping; this code is unnecessary

Reply:
The reason I had this originally was because outside of the class declaration, checking that the method a classmethod is kind of messy (https://stackoverflow.com/questions/19227724/check-if-a-function-uses-classmethod). I can change it to be the messier version on the mode with a note that it's checking for classmethod

Reply:
I don't mind if we just dump the classmethod check entirely, it's not that important.

Reply:
I ended up moving it to the slightly messier check with a comment about what it's doing. Personally I'm kind of attached to it since it came up as someone was doing a migration from subclasses to modes (which I'm guessing will be a common pattern) and forgot to change it from a class method. The error message doesn't reference classmethods and was hard for users to reason about